modules/programs/meli.nix

```
@@ -0,0 +1,222 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  inherit (lib)
+    mkEnableOption
+    mkOption
+    types
+    mkIf
+    ;
+
+  enabledAccounts = lib.attrsets.filterAttrs (
+    name: value: value.meli.enable or false
+  ) config.accounts.email.accounts;
+
+  meliAccounts = (lib.attrsets.mapAttrs (name: value: (mkMeliAccounts name value)) enabledAccounts);
+
+  mkMeliAccounts = (
+    name: account: {
+      root_mailbox = "${config.accounts.email.maildirBasePath}/${account.maildir.path}";
+      format = "Maildir";
+      identity = "${account.address}";
+      display_name = "${account.realName}";
+      subscribed_mailboxes = account.meli.mailboxes;
+      send_mail = mkSmtp account;
+      mailboxes = account.meli.mailboxAliases;
+    }
+  );
+
+  mkSmtp = account: {
+    hostname = account.smtp.host;
+    port = account.smtp.port;
+    auth = {
+      type = "auto";
+      username = account.userName;
+      password = {
+        type = "command_eval";
+        value = lib.strings.concatStringsSep " " account.passwordCommand;
+      };
+    };
+    security = {
+      type =
+        if account.smtp.tls.enable or false then
+          if account.smtp.tls.useStartTls or false then "starttls" else "tls"
+        else
+          "none";
+      danger_accept_invalid_certs = false;
+    };
+    extensions = {
+      PIPELINING = true;
+      CHUNKING = true;
+      PRDR = true;
+      DSN_NOTIFY = "FAILURE";
+    };
+  };
+
+in
+{
+  meta.maintainers = with lib.hm.maintainers; [ munsman ];
+
+  options = {
+    programs.meli = {
+      enable = mkEnableOption "meli email client";
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.meli;
+        description = "meli package to use";
+      };
+
+      extraConfig = mkOption {
+        type = types.lines;
+        default = "";
+        description = "Additional lines to add to meli config.toml";
+      };
+      settings = mkOption {
+        type = types.submodule {
+          options = {
```

```suggestion
          freeformType = (pkgs.formats.toml { }).type;
          options = {
```
We're using a toml generator so we should be able to just accept freeform entry for the user.